lib/osutils/file.go

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2021 PaperCut Software International Pty. Ltd.
+ */
+
+package osutils
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+// FileExists check if a file denoted by path exists, returning true or false.
+func FileExists(path string) bool {
+	_, err := os.Stat(path)
+	return !os.IsNotExist(err)
+}
+
+// CopyFile copies a file from src to dst
+func CopyFile(src, dest string) error {
+	s, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer func() { _ = s.Close() }()
+	d, err := os.Create(dest)
+	if err != nil {
+		return err
+	}
+	if _, err := io.Copy(d, s); err != nil {
+		_ = d.Close()
+		return err
+	}
+	return d.Close()
+}
+
+// WriteFileString writes a string to file denoted by filename with specified permissions.
+func WriteFileString(filename string, data string, perm os.FileMode) error {
+	f, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, perm)
+	if err != nil {
+		return err
+	}
+	_, err = f.WriteString(data)
```

Comment:
we will only return this error if there is no error closing the file?

Reply:
Yep, will add extra if